fvtest/tril/tril/parser.cpp

```diff
@@ -0,0 +1,482 @@
+    } else if (tokenType == Token::HEXINT) {
+        return createIntegerValue(std::strtoull(cstr, NULL, 0));
+    } else {
+        throw TokenFailure("Do not have token type with value" + tokenValue + " as value type");
+    }
+}
+
+
+ASTNodeArg* parseArgList(TokenIter &tokenIt) {
+    ASTNodeArg* head_arg = NULL;
+    ASTNodeArg* head_argList = NULL;
+    ASTValue* head_args_value = NULL;
+    ASTValue* head_args_value_head = NULL;
+
+    try {
+        while (tokenIt.getIt()->getType() != Token::LPAREN && tokenIt.getIt()->getType() != Token::RPAREN) {
+            if (tokenIt.getIt()->getType() == Token::ID && tokenIt.peek()->getType() == Token::BECOMES) {
+                char * node_args_name = new char[tokenIt.getIt()->getValue().size() + 1];
+                strcpy(node_args_name, tokenIt.getIt()->getValue().c_str());
+                if (tokenIt.peek()->getType() == Token::RPAREN) {
+                    head_arg = createNodeArg(node_args_name, buildNodeValue(*(tokenIt.getIt())), NULL);
+                } else {
+                    ++tokenIt;
+                    ++tokenIt;
+                    if (tokenIt.getIt()->getType() == Token::LPAREN_SQ) {
+                        ++tokenIt;
+                        while (tokenIt.getIt()->getType() != Token::RPAREN_SQ) {
+                            if (head_args_value == NULL && tokenIt.getIt()->getType() != Token::COMMA) {
+                                head_args_value = buildNodeValue(*(tokenIt.getIt()));
+                                head_args_value_head = head_args_value;
+                            } else if (head_args_value && tokenIt.getIt()->getType() != Token::COMMA) {
+                                head_args_value->next = buildNodeValue(*(tokenIt.getIt()));
+                                head_args_value = head_args_value->next;
+                            }
+                            ++tokenIt;
+                        }
+                        head_arg = createNodeArg(node_args_name, head_args_value_head, NULL);
+                    } else {
+                        head_arg = createNodeArg(node_args_name, buildNodeValue(*(tokenIt.getIt())), NULL);
+                    }
+                }
+            } else if (tokenIt.getIt()->getType() == Token::DOUBLE_ || tokenIt.getIt()->getType() == Token::INT ||
+                tokenIt.getIt()->getType() == Token::STRING_ || tokenIt.getIt()->getType() == Token::HEXINT ||
+                tokenIt.getIt()->getType() == Token::ZERO || tokenIt.getIt()->getType() == Token::ID) {
+                head_arg = createNodeArg("", buildNodeValue(*(tokenIt.getIt())), NULL);
+            } else {
+                throw TokenFailure("Incorrect type for argument, current value is " + tokenIt.getIt()->getValue());
+            }
+            if (head_argList == NULL) {
+                head_argList = head_arg;
+            } else {
+                appendSiblingArg(head_argList, head_arg);
+            }
+            ++tokenIt;
```

Comment:
I think `++tokenIt` should be the start of the next argument, which could be a single node value like `3.5`, or the argument with specific ID, for example, `return=Int32`, since this is inside `parseArgList` function, and we will stop if meets a `)`.